samples/append_rows_proto2.js

```diff
@@ -0,0 +1,313 @@
+    .FieldDescriptorProto.Type;
+  const mode = require('@google-cloud/bigquery-storage').protos.google.cloud
+    .bigquery.storage.v1.WriteStream.Type;
+
+  async function appendRowsProto2() {
+    /**
+     * If you make updates to the sample_data.proto protocol buffers definition,
+     * run:
+     *   protoc --js_out=import_style=commonjs,binary:. sample_data.proto
+     * from the /samples directory to generate the sample_data_pb2 module.
+     */
+
+    const writeClient = new BigQueryWriteClient();
+
+    /**
+     * TODO(developer): Uncomment the following lines before running the sample.
+     */
+    // projectId = 'my_project';
+    // datasetId = 'my_dataset';
+    // tableId = 'my_table';
+
+    const parent = `projects/${projectId}/datasets/${datasetId}/tables/${tableId}`;
+
+    // So that BigQuery knows how to parse the serialized_rows, create a
+    // protocol buffer representation of your message descriptor.
+    const protoDescriptor = {};
+    protoDescriptor.name = 'SampleData';
+    protoDescriptor.field = [
+      {
+        name: 'bool_col',
+        number: 1,
+        type: type.TYPE_BOOL,
+      },
+      {
+        name: 'bytes_col',
+        number: 2,
+        type: type.TYPE_BYTES,
+      },
+      {
+        name: 'float64_col',
+        number: 3,
+        type: type.TYPE_FLOAT,
+      },
+      {
+        name: 'int64_col',
+        number: 4,
+        type: type.TYPE_INT64,
+      },
+      {
+        name: 'string_col',
+        number: 5,
+        type: type.TYPE_STRING,
+      },
+      {
+        name: 'date_col',
+        number: 6,
+        type: type.TYPE_INT32,
+      },
+      {
+        name: 'datetime_col',
+        number: 7,
+        type: type.TYPE_STRING,
+      },
+      {
+        name: 'geography_col',
+        number: 8,
+        type: type.TYPE_STRING,
+      },
+      {
+        name: 'numeric_col',
+        number: 9,
+        type: type.TYPE_STRING,
+      },
+      {
+        name: 'bignumeric_col',
+        number: 10,
+        type: type.TYPE_STRING,
+      },
+      {
+        name: 'time_col',
+        number: 11,
+        type: type.TYPE_STRING,
+      },
+      {
+        name: 'timestamp_col',
+        number: 12,
+        type: type.TYPE_INT64,
+      },
+      {
+        name: 'int64_list',
+        number: 13,
+        type: type.TYPE_INT64,
+        label: protos.TableFieldSchema.Mode.REPEATED,
+      },
+    ];
+
+    try {
+      // Create a write stream to the given table.
+      let writeStream = {type: mode.PENDING};
+
+      let request = {
+        parent,
+        writeStream,
+      };
+
+      let [response] = await writeClient.createWriteStream(request);
+
+      console.log(`Stream created: ${response.name}`);
+
+      writeStream = response.name;
+
+      // Append data to the given stream.
+      const stream = await writeClient.appendRows();
+
+      stream.on('data', response => {
+        // Response recieved from API.
+        console.log(response);
```

Comment:
I think you'll want to increment something based on how many rows were written.

With a `PENDING` stream, the rows aren't committed to the table until the stream is finalized and then (batch) committed. That means once we know all the expected rows were written we should close the local stream object, make the finalized RPC, and then make the batch commit RPC.